src/main/java/io/stargate/sgv2/jsonapi/service/embedding/DataVectorizer.java

```diff
@@ -43,48 +44,62 @@ public DataVectorizer(EmbeddingService embeddingService, JsonNodeFactory nodeFac
    *
    * @param documents - Documents to be vectorized
    */
-  public void vectorize(List<JsonNode> documents) {
-    int vectorDataPosition = 0;
-    List<String> vectorizeTexts = new ArrayList<>();
-    Map<Integer, Integer> vectorizeMap = new HashMap<>();
-    for (int position = 0; position < documents.size(); position++) {
-      JsonNode document = documents.get(position);
-      if (document.has(DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD)) {
-        if (document.has(DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD)) {
-          throw new JsonApiException(ErrorCode.INVALID_USAGE_OF_VECTORIZE);
-        }
-        final JsonNode jsonNode =
-            document.get(DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD);
-        if (jsonNode.isNull()) {
-          ((ObjectNode) document)
-              .put(DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD, (String) null);
-          continue;
-        }
-        if (!jsonNode.isTextual()) {
-          throw new JsonApiException(ErrorCode.SHRED_BAD_VECTORIZE_VALUE);
-        }
+  public Uni<Boolean> vectorize(List<JsonNode> documents) {
+    try {
+      int vectorDataPosition = 0;
+      List<String> vectorizeTexts = new ArrayList<>();
+      Map<Integer, Integer> vectorizeMap = new HashMap<>();
+      for (int position = 0; position < documents.size(); position++) {
+        JsonNode document = documents.get(position);
+        if (document.has(DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD)) {
+          if (document.has(DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD)) {
+            throw new JsonApiException(ErrorCode.INVALID_USAGE_OF_VECTORIZE);
```

Comment:
Given that this may be called on bunch of documents, it would be good to indicate which document (index) had the problem?